Document adding admonitions through PyMdown (#190)
ntno/mkdocs-terminal: documentation/docs/configuration/extensions/pymdown-extensions.md

  [Highlight]: https://facelessuser.github.io/pymdown-extensions/extensions/highlight/
  [InlineHilite]: https://facelessuser.github.io/pymdown-extensions/extensions/inlinehilite/

> ## Admonitions

Reviewer: let's call the section "Details" since "Admonitions" is its own extension type.  

```suggestion
## Details
```

The [Details] extension can be used to create admonitions (sometimes called callouts). By default, mkdocs terminal supports three admonition types: default, error, and primary. Enable them via `mkdocs.yml`:

> ```yaml

Reviewer: i think we should either add content for the suggested `title` attributes or remove the entries entirely

Author: Have removed the suggested title attributes and instead shown the examples as they would appear if enabled.

markdown_extensions:
  - pymdownx.blocks.details:
      types:

>         - name: 'default'

Reviewer: YAML style change request - can you remove the single quotes from this example?  i think it makes sense when a string has white space but if the string doesn't have whitespace i would prefer to remove them

```suggestion
        - name: default
```

          class: 'terminal-alert'
          title: ''
        - name: 'error'
          class: 'terminal-alert-error'
          title: ''
        - name: 'primary'
          class: 'terminal-alert-primary'
          title: ''
```

> If you wish to use different admonition types, you can change the types above, and then define the custom classes in a [css override](/tile-grid/examples/override-styling).

Reviewer: ```suggestion
If you wish to use different detail types, you can change the types above, and then define the custom classes in a [css override](/tile-grid/examples/override-styling).
```

> If you wish to use different admonition types, you can change the types above, and then define the custom classes in a [css override](/tile-grid/examples/override-styling).

Reviewer: i think we should recommend using the [Mkdocs `extra_css` ](https://www.mkdocs.org/user-guide/configuration/#extra_css) feature to load custom CSS classes rather than the [inline `<style>` block in markdown](https://ntno.github.io/mkdocs-terminal/tile-grid/examples/override-styling/#extra-css).  the inline option is useful for one-off overrides but the `extra_css` option makes more sense when you plan on reusing a CSS chunk throughout the docs site. 

now that i am looking at the https://ntno.github.io/mkdocs-terminal/tile-grid/examples/override-styling/#extra-css page i am realizing i shouldn't have titled the section "Extra CSS" 😅 

if you think the inline example is useful though i am ok with including both options

  [Highlight]: https://facelessuser.github.io/pymdown-extensions/extensions/highlight/
  [InlineHilite]: https://facelessuser.github.io/pymdown-extensions/extensions/inlinehilite/

## Details

> The [Details] extension can be used to create admonitions (sometimes called callouts). By default, mkdocs terminal supports three details types. Enable them via `mkdocs.yml`:

Reviewer: - update to name convention when talking about this particular theme ("Terminal for MkDocs")
- wording update for clarity

```suggestion
The [Details] extension can be used to create admonitions (sometimes called callouts). By default, Terminal for MkDocs supports three styles.  Enable them via `mkdocs.yml`:
```

/// info
This is an info block
///

/// warning
This is a warning
///

/// important
This is important
///

```yaml
markdown_extensions:
  - pymdownx.blocks.details:
      types:
        - name: info 
          class: terminal-alert
          title: Info
        - name: warning
          class: terminal-alert-error
          title: Warning
        - name: important
          class: terminal-alert-primary
          title: Important
```

> If you wish to use different detail types, you can change the types above, and then define the custom classes in a extra_css file.

Reviewer: i think calling each type a "category" is less confusing

```suggestion
* To use different categories, update the `name` attribute.
* To use a custom style, update the `class` attribute to the name of your custom CSS class (remember to load the custom CSS via the [extra_css MkDocs feature]).

  [Details]: https://facelessuser.github.io/pymdown-extensions/extensions/blocks/plugins/details/
  [extra_css MkDocs feature]: https://www.mkdocs.org/user-guide/configuration/#extra_css
```

>   [Details]: https://facelessuser.github.io/pymdown-extensions/extensions/blocks/plugins/details/

Reviewer: can you add a link at the end of this section to the example page?  ty!

```suggestion
  [Details]: https://facelessuser.github.io/pymdown-extensions/extensions/blocks/plugins/details/
  
 See reference for usage:

- [Adding Details]

  [Adding Details]: ../../../elements/details
```
